src/studio/ui/StudioPanel.java

```diff
@@ -2182,14 +2242,19 @@ private void processK4Results(K.KBase r) throws c.K4Exception {
         else {
             // Log that execute was successful
         }
+        tabbedPane.setSelectedIndex(tabbedPane.getTabCount()-1);
     }
     Server server = null;
 
       public void executeK4Query(final String text) {
         final Cursor cursor = textArea.getCursor();
 
         textArea.setCursor(new java.awt.Cursor(java.awt.Cursor.WAIT_CURSOR));
-        tabbedPane.removeAll();
+
+          if(tabbedPane.getTabCount()>=Config.getInstance().getResultTabsCount()) {
```

Comment:
For table, list and dictionary results, we create 2 tabs. I just tried this PR locally and what ends up happening is that if you go over the limit set in the options, 2 new tabs are opened but only one is closed, so you end up with more and more tabs.

Reply:
Thank you